crates/subspace-networking/src/behavior.rs

```diff
@@ -42,30 +46,45 @@ impl Behavior {
             kademlia
         };
 
+        let gossipsub = Gossipsub::new(
+            // TODO: Do we want message signing?
+            MessageAuthenticity::Anonymous,
+            config.gossipsub,
+        )
+        .expect("Correct configuration");
+
         Self {
-            kademlia,
             identify: Identify::new(config.identify),
+            kademlia,
+            gossipsub,
             ping: Ping::default(),
         }
     }
 }
 
 #[derive(Debug)]
 pub(crate) enum Event {
-    Kademlia(KademliaEvent),
     Identify(IdentifyEvent),
+    Kademlia(KademliaEvent),
+    Gossipsub(GossipsubEvent),
     Ping(PingEvent),
 }
 
+impl From<IdentifyEvent> for Event {
```

Comment:
Maybe we could remove those with just derive macro?
Here is [`derive_more::From`](https://jeltef.github.io/derive_more/derive_more/from.html). Also there is [`TryInto`](https://jeltef.github.io/derive_more/derive_more/try_into.html) derive which works with enums.